### DISCUSSION TITLE
compiler should check int overflow

### DISCUSSION THREAD

**Discussion by MaxGraey** [vlang/v#11575] [Ideas]
```go
fn main() {
	foo := [1; - 1]
	println(foo)
}
```
```
V panic: malloc <0
```
Also:
```go
foo := [1; 4294967295]
// or
foo := [1; 2147483647]
// or
foo := [1; 1073741823]
// or
foo := [1; 573741823]
...
```
Output:
```
V panic: malloc <0
```

**Comment by medvednikov** [comment]
What's wrong with the panic when you try to create an array with -1 elements?

**Comment by MaxGraey** [comment]
@medvednikov It should be compile time error in this case. Or `-1` can cast to `unsigned int` and interpret as `4294967295`.
Also `[1; 1073741823]` is not negative value but also cause to `V panic: malloc <0`

**Comment by medvednikov** [comment]
There's no compile time check of integer literals.

It's not very critical, so I don't want to spend time on this right now. It will be taken care of after the language is open-sourced.

**Comment by AurelienFT** [comment]
Update the issue with new syntax : 
```
fn main() {
	foo := [1].repeat(-1)
    println(foo)
}
```
Ouput (at run time) : 
```
V panic: array.repeat: count is negative (count == -1)
??:0:                                         | ./main(+0x2d47) [0x564632f81d47]
??:0:                                         | ./main(+0x1aab) [0x564632f80aab]
??:0:                                         | ./main(+0xa713) [0x564632f89713]
??:0:                                         | ./main(+0xab5e) [0x564632f89b5e]
??:0:                                         | /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f452d8381e3]
??:0:                                         | ./main(+0x13ee) [0x564632f803ee]
```

**Comment by JalonSolov** [comment]
While the compiler _could_ check when it's a literal, it really should be checked in the repeat() function, possibly in addition, in case you pass a variable instead of a constant.

Note that this is still failing in the latest version of V at this time, `V 0.1.29 4774c89`
